File: src/main/java/de/take_weiland/mods/commons/internal/PacketInventoryName.java
package de.take_weiland.mods.commons.internal;

import de.take_weiland.mods.commons.inv.Containers;
import de.take_weiland.mods.commons.net.*;
import de.take_weiland.mods.commons.util.JavaUtils;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.inventory.IInventory;
import net.minecraft.util.text.ITextComponent;
import net.minecraftforge.fml.relauncher.Side;

public final class PacketInventoryName implements Packet {

    private final int windowId;
    private final int invIdx;
    private final ITextComponent name;

    public PacketInventoryName(int windowId, int invIdx, ITextComponent name) {
        this.windowId = windowId;
        this.invIdx = invIdx;
        this.name = name;
    }

    PacketInventoryName(MCDataInput in) {
        windowId = in.readByte();
        invIdx = in.readByte();
        name = ITextComponent.Serializer.jsonToComponent(in.readString());
    }

    @Override
    public void writeTo(MCDataOutput out) {
        out.writeByte(windowId);
        out.writeByte(invIdx);
        out.writeString(ITextComponent.Serializer.componentToJson(name));
    }

    @PacketHandler.ReceivingSide(Side.CLIENT)
    void handle(EntityPlayer player) {
        if (player.openContainer.windowId == windowId) {
            // TODO
            IInventory inv = JavaUtils.get(Containers.getInventories(player.openContainer).asList(), invIdx);
//            if (inv instanceof NameableInventory) {
//                ((NameableInventory) inv).setCustomName(name);
//            }
        }
    }
}

File: src/main/java/de/take_weiland/mods/commons/internal/PacketInventoryName.kt
package de.take_weiland.mods.commons.internal

import de.take_weiland.mods.commons.net.packet.mod.Packet
import de.take_weiland.mods.commons.net.readString
import de.take_weiland.mods.commons.net.readVarInt
import de.take_weiland.mods.commons.net.writeString
import io.netty.buffer.ByteBuf
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.text.ITextComponent
import net.minecraft.util.text.TextComponentString
import net.minecraftforge.fml.relauncher.Side

class PacketInventoryName : Packet {

    private val windowId: Int
    private val slotIndex: Int
    private val name: ITextComponent

    constructor(windowId: Int, invIdx: Int, name: ITextComponent) {
        this.windowId = windowId
        this.slotIndex = invIdx
        this.name = name
    }

    internal constructor(input: ByteBuf) {
        windowId = input.readByte().toInt()
        slotIndex = input.readVarInt()
        name = ITextComponent.Serializer.jsonToComponent(input.readString()) ?: TextComponentString("")
    }

    override fun write(buf: ByteBuf) {
        buf.writeByte(this.windowId)
        buf.writeByte(this.slotIndex)
        buf.writeString(ITextComponent.Serializer.componentToJson(this.name))
    }

    override fun receive(side: Side, player: EntityPlayer) {
        if (player.openContainer.windowId == windowId) {
            // TODO
            val inv = player.openContainer.inventorySlots.getOrNull(slotIndex)?.let { inv ->

            }
            //            if (inv instanceof NameableInventory) {
            //                ((NameableInventory) inv).setCustomName(name);
            //            }
        }
    }

}
